### PR TITLE
Add support for cascade layers

### PR DESCRIPTION
**Pull request by Prestaul**
## Description
User-agent and polyfilled styles should always have lowest priority, but if a consumer uses cascade layers (i.e. @layer) then polyfilled popover styles have highest priority and overwrite other styles unless they are also in a layer. This change puts polyfilled styles into a layer so that consumers can control layer order.

## Related Issue(s)
#175 (original issue and discussion)

## Steps to test/reproduce
Place a style that sets the background or color of any popover element into any `@layer` block. That style will only apply correctly after this change.

### CODE REVIEW

**Comment by Prestaul** [comment]
I apologize for the stacked PR. This seemed the best way to avoid conflicts with my other PR, but I'm happy to break this up and just deal with conflicts and rebasing if one lands.

The main issue is in the `index.html` where each PR is adding a new popover and button for testing. The existing pattern of just incrementing a number in the id (i.e. `id="popover12`) doesn't lend itself to concurrent changes.

File: README.md [hunk 1]
```diff
@@ -142,6 +142,11 @@ some caveats which will need accommodations:
     now is to add `!important` to conflicting properties in your `:host` rule.
     See [#147](https://github.com/oddbird/popover-polyfill/issues/147) for more.
 
+- When supported, the polyfill creates a cascade layer named `popover-polyfill`.
+  If your styles are not in layers then this should have no impact. If your
+  styles do use layers, you'll need to ensure the polyfill layer is declared
+  first. (e.g. `@layer popover-polyfill, other, layers;`)
+
```
Comment (keithamus) [inline README.md:149]: (Non blocking question) Can layers be re-used? Would it make sense to have `@layer polyfill {}` instead, in the hopes that it can become the defacto layer to put CSS polyfills in?

File: src/popover.ts [hunk 2]
```diff
@@ -34,10 +34,12 @@ function patchSelectorFn<K extends string>(
 }
 
 const nonEscapedPopoverSelector = /(^|[^\\]):popover-open\b/g;
+const hasLayerSupport = Boolean(window.CSSLayerBlockRule);
```
Comment (keithamus) [inline src/popover.ts:37]: Non blocking nit:

```suggestion
const hasLayerSupport = typeof window.CSSLayerBlockRule === 'function';
```

File: tests/edge-cases.spec.ts [hunk 3]
```diff
@@ -20,3 +20,10 @@ test("removing an autoPopover from document doesn't crash page", async ({
   ).toBeUndefined();
   await expect(popover2).toBeVisible();
 });
+
+test('styles in layers are winning over polyfilled styles', async ({
+  page,
+}) => {
+  const popover = (await page.locator('#popover13')).nth(0);
+  await expect(popover).toHaveCSS('background-color', 'rgb(208, 13, 30)');
+});
```
Comment (keithamus) [inline tests/edge-cases.spec.ts:29]: Nice test :+1:

**Review by keithamus — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (Prestaul) [reply README.md:149]: That's actually a really good call. Yes, they can be extended and they can have hierarchy. Maybe `polyfill.popover` would be best. That gives consumers the chance to set a layer order including `polyfill` alone, but we can also set an order for specific polyfills, if needed.

@shannonmoeller also had thoughts about later naming conventions for public packages.

File: tests/edge-cases.spec.ts (hunk 3 above)
Comment (jamesnw) [reply tests/edge-cases.spec.ts:29]: Per this [blog post](https://playwrightsolutions.com/what-the-hex-or-how-i-check-colors-with-playwright), Playwright does expect it to be rgb. I'm also seeing `rgb()` in Playwright's own tests, but I'm not seeing this documented.

Comment (keithamus) [reply tests/edge-cases.spec.ts:29]: I'm pretty sure this must be `getComputedStyle()`, so the browser is doing it.

Comment (Prestaul) [reply tests/edge-cases.spec.ts:29]: Looks like it is spec'd that way for `getComputedStyle()`, so this is probably okay. https://stackoverflow.com/a/67006298/5628

File: README.md (hunk 1 above)
Comment (shannonmoeller) [reply README.md:149]: I thought it might make sense to mirror npm or repo namespacing in the layers to ensure uniqueness and manageability. Maybe something like one of these:

- `@layer npm.oddbird.popover-polyfill`
- `@layer github.oddbird.popover-polyfill`
- `@layer vendor.oddbird.popover-polyfill`

Comment (Prestaul) [reply README.md:149]: Ah, I'm glad I asked, but I think that probably won't work because it would force all the `npm.*` or `oddbird.*` styles to be grouped together in the layer order. It seems reasonable to me to group all `polyfills.*` together, but even that might be presumptuous without some namespacing... 

Does anyone have thoughts about `oddbird-polyfills.popover` vs `polyfills.popover` to avoid potential naming conflicts?

Comment (shannonmoeller) [reply README.md:149]: Great points. I think `oddbird-polyfills.popover` is pretty unique. Though `polyfills.popover` seems good too. If people are including more than one popover polyfill on the page which would conflict with that I think they have a bigger issue to deal with.

Comment (mirisuzanne) [reply README.md:149]: Whatever we call our internal layers, consuming authors can create their own groupings and avoid name conflicts by nesting on their end. I don't think we should worry about trying to do those things for them. I'd say `popover-polyfill` on it's own is probably plenty, and we don't even need the sublayer I suggested in the issue.

Layer-naming-by-vendor/source conflates two things that aren't really related in my mind. But whatever we do, authors can pull these styles into an `oddbird` layer if they choose to.

Comment (mirisuzanne) [reply README.md:149]: Oh that's true in this case. I'm not used to injecting styles on someone else's site 😅 - but yeah, I think the simpler solution still makes sense.

**Review by jgerigmeyer — APPROVED** [review]
Thanks!

> The existing pattern of just incrementing a number in the id (i.e. id="popover12) doesn't lend itself to concurrent changes.

Yes -- we're working on creating a better demo page.
